### PR TITLE
create skeleton java springboot app for accounts-filing-api

### PR DESCRIPTION
**Pull request by myadavch**
Adding authentication, logging and other dependencies in the api skeleton.

### CODE REVIEW

File: Tiltfile.dev [hunk 1]
```diff
@@ -0,0 +1,26 @@
+local_resource(
+  name = 'dev:account-filing-api',
+  cmd = 'mvn compile',
+  deps = ['src/main']
+)
+
+custom_build(
+  ref = '169942020521.dkr.ecr.eu-west-1.amazonaws.com/local/account-filing-api',
```
Comment (wdhda-ch) [inline Tiltfile.dev:8]: `/account-filing-api` is missing an "s"?

Should it be `/accounts-filing-api` ?

File: Tiltfile.dev [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    ),
+    restart_container()
+  ],
+  deps = ['./target/classes']
+)
+
+load('ext://uibutton', 'cmd_button')
+
+cmd_button('account-filing-api:toggle',
+           argv=['sh', '-c', 'if [ "$(tilt status --resource account-filing-api)" = "enabled" ]; then tilt disable my_resource; else tilt enable account-filing-api; fi'],
```
Comment (wdhda-ch) [inline Tiltfile.dev:23]: Should it be /accounts-filing-api ?

File: routes.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+app_name: account-filing-api
```
Comment (wdhda-ch) [inline routes.yaml:1]: Should it be accounts-filing-api ?

File: src/main/java/uk/gov/companieshouse/account/filing/AccountFilingApiApplication.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package uk.gov.companieshouse.account.filing;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class AccountFilingApiApplication {
+
+	public static void main(String[] args) {
+		SpringApplication.run(AccountFilingApiApplication.class, args);
```
Comment (wdhda-ch) [inline src/main/java/uk/gov/companieshouse/account/filing/AccountFilingApiApplication.java:10]: Should it be `AccountsFilingApiApplication` ?

File: Tiltfile.dev (hunk 1 above)
Comment (myadavch) [reply Tiltfile.dev]: Done

File: Tiltfile.dev (hunk 2 above)
Comment (myadavch) [reply Tiltfile.dev]: Done

File: routes.yaml (hunk 3 above)
Comment (myadavch) [reply routes.yaml]: Done

File: src/main/java/uk/gov/companieshouse/account/filing/AccountFilingApiApplication.java (hunk 4 above)
Comment (myadavch) [reply src/main/java/uk/gov/companieshouse/account/filing/AccountFilingApiApplication.java]: Done

**Review by amartin7663 — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -1,2 +1,13 @@
-# accounts-filing-api
-A backend service allowing users to submit accounts in zip format.
+# account-filing-api
```
Comment (amartin7663) [inline README.md:1]: Still have account not accounts in the read me

**Review by amartin7663 — CHANGES_REQUESTED** [review]

File: Tiltfile.dev [hunk 6]
```diff
@@ -0,0 +1,26 @@
+local_resource(
+  name = 'dev:account-filing-api',
```
Comment (amartin7663) [inline Tiltfile.dev:2]: accounts here

File: README.md [hunk 7]
```diff
@@ -1,2 +1,13 @@
 # accounts-filing-api
-A backend service allowing users to submit accounts in zip format.
+A service to manage user's account for all their filings.This application is written using the [Spring Boot](http://projects.spring.io/spring-boot/) Java framework.
```
Comment (amartin7663) [inline README.md:2]: I think I prefer the original text so something like this
```suggestion
A backend service allowing users to submit accounts in zip format. This application is written using the [Spring Boot](http://projects.spring.io/spring-boot/) Java framework.
```

**Review by amartin7663 — APPROVED** [review]

**Pull request merged**